Vagrantfile

```diff
@@ -74,23 +30,23 @@ else
   config_dir = guest_config_dir
 end
 
-if provisioner == :ansible && ansible_version < ansible_version_min
-  raise Vagrant::Errors::VagrantError.new, "You must update Ansible to at least #{ansible_version_min} to use this version of Drupal VM."
-end
+# Verify version requirements.
+require_ansible_version ">= #{vconfig['drupalvm_ansible_version_min']}"
+Vagrant.require_version ">= #{vconfig['drupalvm_vagrant_version_min']}"
+
+Vagrant.configure('2') do |config|
+  # Set the name of the VM. See: http://stackoverflow.com/a/17864388/100134
+  config.vm.define vconfig['vagrant_machine_name']
 
-Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
   # Networking configuration.
   config.vm.hostname = vconfig['vagrant_hostname']
-  if vconfig['vagrant_ip'] == '0.0.0.0' && Vagrant.has_plugin?('vagrant-auto_network')
-    config.vm.network :private_network, ip: vconfig['vagrant_ip'], auto_network: true
-  else
-    config.vm.network :private_network, ip: vconfig['vagrant_ip']
-  end
+  config.vm.network :private_network,
+    ip: vconfig['vagrant_ip'],
+    auto_network: vconfig['vagrant_ip'] == '0.0.0.0' && Vagrant.has_plugin?('vagrant-auto_network')
 
-  if !vconfig['vagrant_public_ip'].empty? && vconfig['vagrant_public_ip'] == '0.0.0.0'
-    config.vm.network :public_network
-  elsif !vconfig['vagrant_public_ip'].empty?
-    config.vm.network :public_network, ip: vconfig['vagrant_public_ip']
+  if vconfig['vagrant_public_ip']
+    config.vm.network :public_network,
+      ip: vconfig['vagrant_public_ip'] != '0.0.0.0' ? vconfig['vagrant_public_ip'] : nil
```

Comment:
My personal preference but I know some prefer to avoid ternary expressions so let me know if you want me to revert it. Also I'm only assuming `nil` will be the same as not setting the option at all. Should verify that.